brew should create the dist/formula.rb file when --skip-publish is set

This broke something for me. It no longer generates the formulas in the dist folder for me be debug the configuration 😭

_Originally posted by @kreempuff in https://github.com/goreleaser/goreleaser/issues/2193#issuecomment-833001315_

@kreempuff Sure. I would run `release --rm-dist --skip-publish --snapshot`  and would see the following

<details><summary> Sample output</summary>

```
# ... 

 • publishing
      • blobs
      • http upload
      • custom publisher
      • artifactory
      • docker images
         • pipe skipped              error=publishing is disabled
      • docker manifests
         • pipe skipped              error=publishing is disabled
      • snapcraft packages
         • pipe skipped              error=publishing is disabled
      • github/gitlab/gitea releases
         • pipe skipped              error=publishing is disabled
      • homebrew tap formula
   ⨯ release failed after 57.40s error=homebrew tap formula: failed to publish artifacts: no linux/macos archives found
make: *** [local_release] Error 1
```
</details>

Note the status code would also be `2` so would fail along a CICD

--- a/internal/pipe/brew/brew.go
+++ b/internal/pipe/brew/brew.go
@@ -48,9 +48,6 @@ func (Pipe) String() string {
 
 // Publish brew formula.
 func (Pipe) Publish(ctx *context.Context) error {
-	if ctx.SkipPublish {
-		return pipe.ErrSkipPublishEnabled
-	}
 	// we keep GitHub as default for now, in line with releases
 	if string(ctx.TokenType) == "" {
 		ctx.TokenType = context.TokenTypeGitHub

--- a/internal/pipe/brew/brew_test.go
+++ b/internal/pipe/brew/brew_test.go
@@ -10,7 +10,6 @@ import (
 
 	"github.com/goreleaser/goreleaser/internal/artifact"
 	"github.com/goreleaser/goreleaser/internal/client"
-	"github.com/goreleaser/goreleaser/internal/pipe"
 	"github.com/goreleaser/goreleaser/internal/testlib"
 	"github.com/goreleaser/goreleaser/pkg/config"
 	"github.com/goreleaser/goreleaser/pkg/context"
@@ -852,16 +851,6 @@ func TestRunTokenTypeNotImplementedForBrew(t *testing.T) {
 	})
 	client := &DummyClient{NotImplemented: true}
 	require.Equal(t, ErrTokenTypeNotImplementedForBrew{TokenType: "gitea"}, doRun(ctx, ctx.Config.Brews[0], client))
-}
-
-func TestRunPipe_SkipWhenPublishFalse(t *testing.T) {
-	ctx := context.New(config.Project{})
-	ctx.SkipPublish = true
-
-	require.NoError(t, Pipe{}.Default(ctx))
-	err := Pipe{}.Publish(ctx)
-	require.True(t, pipe.IsSkip(err))
-	require.EqualError(t, err, pipe.ErrSkipPublishEnabled.Error())
 }
 
 func TestDefault(t *testing.T) {

